Player.java

```diff
@@ -17,23 +17,27 @@ public Player( String name, int dmg, int hp, double as, int xppl, int dmgipl, in
     }
 
     public void attack(Unit enemy) {
-        if (enemy.isAlive() && this.isAlive()) {
-            enemy.setHP(enemy.getHp()-this.getDMG());
+        if(enemy.hp<this.dmg){
+            this.xp += enemy.hp;
+            super.attack(enemy);
+            this.lvlup();
+        } else {
+            super.attack(enemy);
+            this.xp +=this.getDMG();
+            this.lvlup();
         }
-        this.xp = this.xp+this.getDMG();
-        this.lvlup();
     }
 
     public boolean isAlive() {
         return super.isAlive();
     }
 
     public void lvlup(){
-        while(this.xp>this.xp_per_level){
-            this.xp=this.xp-this.xp_per_level;
-            setHP(getHp()+hp_increase_per_level);
-            setDMG(getDMG()+dmg_increase_per_level);
-            setAs(getAs()*cooldown_multiplier_per_level);
+        while(this.xp>=this.xp_per_level){
+            this.xp-=this.xp_per_level;
+            this.hp+=this.hp_increase_per_level;
```

Comment:
ezt a masik csapatnal lehet beneztem, de a feladatkiirasban amugy az volt, hogy ilyenkor ne csak megnojon az aktualis hp, hanem fel is menjen max-ra.